source/Applications/Advanced/MaskPointCloud/MaskPointCloud.cpp

```diff
@@ -0,0 +1,224 @@
+        cv::waitKey(0);
+    }
+
+    pcl::PointCloud<pcl::PointXYZRGB> convertToPCLPointCloud(const Zivid::PointCloud &pointCloud)
+    {
+        const auto data = pointCloud.copyData<Zivid::PointXYZColorRGBA>();
+
+        std::cout << "Creating PCL point cloud structure" << std::endl;
+        pcl::PointCloud<pcl::PointXYZRGB> pointCloudPCL;
+
+        std::cout << "Filling in point cloud data" << std::endl;
+        pointCloudPCL.width = pointCloud.width();
+        pointCloudPCL.height = pointCloud.height();
+        pointCloudPCL.is_dense = false;
+        pointCloudPCL.points.resize(pointCloudPCL.width * pointCloudPCL.height);
+
+        for(size_t i = 0; i < pointCloudPCL.points.size(); ++i)
+        {
+            pointCloudPCL.points[i].x = data(i).point.x; // NOLINT(cppcoreguidelines-pro-type-union-access)
+            pointCloudPCL.points[i].y = data(i).point.y; // NOLINT(cppcoreguidelines-pro-type-union-access)
+            pointCloudPCL.points[i].z = data(i).point.z; // NOLINT(cppcoreguidelines-pro-type-union-access)
+            pointCloudPCL.points[i].r = data(i).color.r; // NOLINT(cppcoreguidelines-pro-type-union-access)
+            pointCloudPCL.points[i].g = data(i).color.g; // NOLINT(cppcoreguidelines-pro-type-union-access)
+            pointCloudPCL.points[i].b = data(i).color.b; // NOLINT(cppcoreguidelines-pro-type-union-access)
+        }
+        return pointCloudPCL;
+    }
+
+} // namespace
+
+int main()
+{
+    try
+    {
+        Zivid::Application zivid;
+
+        std::string fileName = std::string(ZIVID_SAMPLE_DATA_DIR) + "/Zivid3D.zdf";
+        std::cout << "Reading ZDF frame from file: " << fileName << std::endl;
+        const auto frame = Zivid::Frame(fileName);
+        const auto pointCloud = frame.pointCloud();
+        std::cout << "Loaded " << pointCloud.size() << " data points" << std::endl;
+
+        const int pixelsToDisplay = 300;
+        std::cout << "Generating binary mask of central " << pixelsToDisplay << " x " << pixelsToDisplay << "pixels."
+                  << std::endl;
+        const int height = pointCloud.height();
+        const int width = pointCloud.width();
+        const int heightMin = (height - pixelsToDisplay) / 2;
+        const int heightMax = (height + pixelsToDisplay) / 2;
+        const int widthMin = (width - pixelsToDisplay) / 2;
+        const int widthMax = (width + pixelsToDisplay) / 2;
+        cv::Mat mask = cv::Mat::zeros(height, width, CV_8U);
+        cv::rectangle(mask,
+                      cv::Point(widthMin, heightMin),
+                      cv::Point(widthMax, heightMax),
+                      cv::Scalar(255, 255, 255),
+                      -1);
+
+        std::cout << "Masking point cloud" << std::endl;
+        const pcl::PointCloud<pcl::PointXYZRGB> maskedPointCloud = maskPointCloud(pointCloud, mask);
+
+        std::cout << "Converting to PCL point cloud" << std::endl;
+        const auto pclPointCloud = convertToPCLPointCloud(pointCloud);
```

Comment:
In other samples, we use the `pointCloudPCL` style. Please use that. Then, the other one can be called `maskedPointCloudPCL`.